R/logger.R

```diff
@@ -29,12 +29,38 @@ setup_logger <- function(output_dir, log_name) {
 
 #' @title Add a basic default file logger for a newly processed file.
 #'
-#' @param output_root Output root directory for the current process.
 #' @param logfile The name of the log file.
-setup_file_logger <- function(output_root, logfile) {
+#' @param output_root Output root directory for the current process.
+#'
+#' @return returns the loggers that where previously set for usage with with_
+setup_file_logger <- function(logfile, output_root) {
+    loggers <- getLoggers()
+    clearLoggers()
     logFileName <- file.path(output_root, "logs", paste0(logfile, ".log"))
     if (file.exists(logFileName)) {
         file.remove(logFileName)
     }
     addDefaultFileLogger(logFileName, name = logfile)
+    loggers
 }
+
+
+
+#' Temporary File Logger
+#'
+#' Temporarily change logging to exclusively log to specific log file
+#'
+#' @param new New filename for the logfile
+#' @param code Code to be executed
+#' @param output_root Root output folder from where to construct the log path
+
+#' @export
+with_file_logger <- withr::with_(
+    setup_file_logger,
```

Comment:
As `withr` is a very special package and at least I was not familiar with it, it is always a good idea to explicitly state the parameter names, in this case `set=` and `reset=`